gcloud/datastore/__init__.py

```diff
@@ -69,6 +69,50 @@
 _DATASET_ENV_VAR_NAME = 'GCLOUD_DATASET_ID'
 
 
+def _find_true_dataset_id(dataset_id, connection=None):
```

Comment:
RE: "know about", if they insert an entity, that will be the dataset ID returned by a `get`, hence the use of the fuzzy comparison in the regression test.

Also, by leaving this out we have previously accidentally introduced bugs like #447